test_runner/fixtures/neon_fixtures.py

```diff
@@ -3708,10 +3741,36 @@ def clear_shared_buffers(self, cursor: Optional[Any] = None):
 
         Might also clear LFC.
         """
+        file_cache_size_limit = 0
         if cursor is not None:
             cursor.execute("select clear_buffer_cache()")
+            cursor.execute("SHOW neon.file_cache_size_limit")
+            res = cursor.fetchone()
+            assert res, "Cannot get neon.file_cache_size_limit"
+            file_cache_size_limit = res[0]
+            if file_cache_size_limit == 0:
+                return
+            cursor.execute("ALTER SYSTEM SET neon.file_cache_size_limit=0")
+            cursor.execute("SELECT pg_reload_conf()")
+            cursor.execute(f"ALTER SYSTEM SET neon.file_cache_size_limit='{file_cache_size_limit}'")
         else:
             self.safe_psql("select clear_buffer_cache()")
+            file_cache_size_limit = self.safe_psql_scalar(
+                "SHOW neon.file_cache_size_limit", log_query=False
+            )
+            if file_cache_size_limit == 0:
+                return
+            self.safe_psql("ALTER SYSTEM SET neon.file_cache_size_limit=0")
+            self.safe_psql("SELECT pg_reload_conf()")
+            self.safe_psql(f"ALTER SYSTEM SET neon.file_cache_size_limit='{file_cache_size_limit}'")
+            self.safe_psql("SELECT pg_reload_conf()")
```

Comment:
Why do we need to have extra `SELECT pg_reload_conf()` here and not in the if-block?

Reply:
We need to set `neon.file_cache_size_limit` to 0 to clear the cache.
Then we need to call `pg_reload_conf` to apply this setting.
And then we need to return this setting.

Reply:
Added gp_reload_conf in the if block as well